efs/main.tf

```diff
@@ -32,3 +32,117 @@ resource "aws_efs_file_system_policy" "file_system_policy" {
   file_system_id = aws_efs_file_system.file_system.id
   policy         = templatefile("${path.module}/templates/${var.policy}.json.tpl", { file_system_arn = aws_efs_file_system.file_system.arn })
 }
+
+resource "aws_security_group" "mount_target_sg" {
+  name        = "mount-target-security-group"
+  description = "Mount target security group"
+  vpc_id      = data.aws_vpc.current.id
+
+  ingress {
+    description     = "EFS"
+    from_port       = 2049
+    to_port         = 2049
+    protocol        = "tcp"
+    security_groups = [aws_security_group.ecs_run_efs.id, aws_security_group.allow_efs_lambda.id]
+  }
+
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    var.common_tags,
+    map("Name", "mount-target-outbound-only")
+  )
+}
+
+resource "aws_security_group" "ecs_run_efs" {
+  name        = "allow-ecs-mount-efs"
+  description = "Allow ECS to mount EFS volume"
+  vpc_id      = data.aws_vpc.current.id
+
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    var.common_tags,
+    map("Name", "allow-ecs-mount-efs")
+  )
+}
+
+resource "aws_security_group" "allow_efs_lambda" {
+  name        = "allow-efs"
+  description = "Allow EFS inbound traffic"
+  vpc_id      = data.aws_vpc.current.id
+
+  ingress {
+    from_port = 2049
```

Comment:
Can this be a variable so it's clear that this has to be the same as in the mount target? (Assuming that's true.)

Reply:
This was wrong, the lambda security group doesn't need incoming on 2049, I've taken it out. Same with the ecs security group.